pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -386,3 +386,114 @@ func createEcDoubleSlopeV1Hinter(resolver hintReferenceResolver) (hinter.Hinter,
 
 	return newEcDoubleSlopeV1Hint(point), nil
 }
+
+
+func newEcDoubleAssignNewXV1(slope, point hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "EcDoubleAssignNewXV1",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+
+			//> slope = pack(ids.slope, PRIME)
+			//> x = pack(ids.point.x, PRIME)
+			//> y = pack(ids.point.y, PRIME)
+
+			//> value = new_x = (pow(slope, 2, SECP_P) - 2 * x) % SECP_P
+
+			slopeAddr, err := slope.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+			slopeMemoryValues, err := hinter.GetConsecutiveValues(vm, slopeAddr, int16(3))
+			if err != nil {
+				return err
+			}
+
+			pointAddr, err := point.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+			pointMemoryValues, err := hinter.GetConsecutiveValues(vm, pointAddr, int16(6))
+			if err != nil {
+				return err
+			}
+
+			// [d0, d1, d2]
+			var slopeValues [3]*fp.Element
+			// [x.d0, x.d1, x.d2]
+			var pointXValues [3]*fp.Element
+			// [y.d0, y.d1, y.d2]
+			var pointYValues [3]*fp.Element
+
+			for i := 0; i < 3; i++ {
+				slopeValue, err := slopeMemoryValues[i].FieldElement()
+				if err != nil {
+					return err
+				}
+				slopeValues[i] = slopeValue
+
+				pointXValue, err := pointMemoryValues[i].FieldElement()
+				if err != nil {
+					return err
+				}
+				pointXValues[i] = pointXValue
+			}
+
+			for i := 3; i < 6; i++ {
+				pointYValue, err := pointMemoryValues[i].FieldElement()
+				if err != nil {
+					return err
+				}
+				pointYValues[i-3] = pointYValue
```

Comment:
Maybe instead of new loop, try to put this assignment in the loop above this one, and go for ```pointYValue, err := pointMemoryValues[i+3].FieldElement()```

Reply:
Great idea!

Reply:
Done! ✅